### DISCUSSION TITLE
Dont work switch_inline_query_current_chat button

### DISCUSSION THREAD

**Discussion by 777petruh** [aso14/DTelegram#4] [Q&A]
Good afternoon again. I found another error. Or maybe I don't understand something. I'm creating a switch_inline_query_current_chat button. And it won’t work until I remove everything unnecessary in Option. As I see it, if the fields are not written in the code, then they should not be in the code sent to the server. There are a lot of empty fields. Which shouldn't exist. When using them, the switch_inline_query_current_chat button does not work.

``
reply_markup={"inline_keyboard":[[{"text":"\u041E\u0442\u043E\u0441\u043B\u0430\u0442\u044C \u0421\u041C\u0421","url":"","callback_data":"","web_app":null,"login_url":null,"switch_inline_query":"","switch_inline_query_current_chat":"\/start_new","switch_inline_query_chosen_chat":null,"pay":false}]]}
{"inline_keyboard":[[{"text":"\u041E\u0442\u043E\u0441\u043B\u0430\u0442\u044C \u0421\u041C\u0421","url":"","callback_data":"","web_app":null,"login_url":null,"switch_inline_query":"","switch_inline_query_current_chat":"\/start_new","switch_inline_query_chosen_chat":null,"pay":false}]]}
``

```
    LButtonList := TList<TTelegramInlineKeyBoardButton>.Create;
    var LButton1  := TTelegramInlineKeyBoardButton.Create;
        LButton1.switch_inline_query_current_chat:='/start_new';
        LButton1.text := 'Отослать СМС';
        LButtonList.Add(LButton1);

    var Response := TelegramBot1.GenInlineKeyBoard(LButtonList,1);

    var Option := TStringList.Create;
      Option.Add('reply_markup='+Response);
/// Need to add this for it to work
    Option.Text:=StringReplace(Option.Text,'"url":"","callback_data":"","web_app":null,"login_url":null,"switch_inline_query":"",','',[rfReplaceAll]);
    Option.Text:=StringReplace(Option.Text,',"switch_inline_query_chosen_chat":null,"pay":false','',[rfReplaceAll]);

    TelegramBot1.SendMessage(Update.result[i].message.from.id.ToString,'Text',Option);

```

**Comment by aso14** [comment]
Let's me check it
